src/CoreWCF.Http/src/CoreWCF/Channels/HttpRequestContext.cs

```diff
@@ -425,22 +434,30 @@ protected override Task OnCloseAsync(CancellationToken token)
                 //}
             }
 
-            private async Task<ServiceSecurityContext> CreateSecurityContextAsync(IIdentity identity)
+            private async Task<ServiceSecurityContext> CreateSecurityContextAsync(IPrincipal principal)
             {
-                if (identity is WindowsIdentity wid)
+                if (principal.Identity is WindowsIdentity wid)
                 {
                     WindowsSecurityTokenAuthenticator tokenAuthenticator = new WindowsSecurityTokenAuthenticator();
                     SecurityToken windowsToken = new WindowsSecurityToken(wid);
                     ReadOnlyCollection<IAuthorizationPolicy> authorizationPolicies = await tokenAuthenticator.ValidateTokenAsync(windowsToken);
                     return new ServiceSecurityContext(authorizationPolicies);
                 }
-                else if (identity is GenericIdentity gid)
+                else if (principal.Identity is GenericIdentity gid)
                 {
                     WindowsSecurityTokenAuthenticator tokenAuthenticator = new WindowsSecurityTokenAuthenticator();
                     SecurityToken genericToken = new GenericIdentitySecurityToken(gid, SecurityUniqueId.Create().Value);
                     ReadOnlyCollection<IAuthorizationPolicy> authorizationPolicies = await tokenAuthenticator.ValidateTokenAsync(genericToken);
                     return new ServiceSecurityContext(authorizationPolicies);
                 }
+                else if (principal.Identity is ClaimsIdentity cid)
+                {
+                    Fx.AssertAndThrow(cid.IsAuthenticated, "ClaimsPrincipal should be authenticated");
+                    AuthorizationContext authorizationContext = AuthorizationContext.CreateDefaultAuthorizationContext(null);
```

Comment:
Can this assert be hit? ValidateAuthentication already checks IsAuthenticated so if that returns Unauthorized, this code won't be hit.

Reply:
indeed i removed it here https://github.com/CoreWCF/CoreWCF/pull/850/commits/64ed1127e474fddb939539fc8bb780fa003ce018